app/lib/core/models/atr_identifier.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+
+@immutable
+class AtrIdentifier {
+  final String userId;
+  final String atrDocumentId;
+  final bool isComplete;
+
+  AtrIdentifier({
+    @required this.userId,
+    this.atrDocumentId,
+    this.isComplete = false,
+  });
+
+  factory AtrIdentifier.fromJSON(
+      Map<String, dynamic> json, String atrDocumentId) {
+    return AtrIdentifier(
+      userId: json['userId'],
+      isComplete: json['isComplete'],
+      atrDocumentId: atrDocumentId,
```

Comment:
Isn't the document ID in the JSON?

Reply:
We don't need to pass atrId because it's sole purpose is to identify which atr needs to be updated. We use this behind the scene. This idea is based on [FilledStack](https://www.youtube.com/watch?v=1PhAPWzGaM4&t=1257s&ab_channel=FilledStacks) tutorial. If you'd like to see it, start on 18:40